src/main/java/com/cognifide/aemrules/htl/visitors/HtlScanner.java

```diff
@@ -0,0 +1,172 @@
+/*-
+ * #%L
+ * AEM Rules for SonarQube
+ * %%
+ * Copyright (C) 2015-2018 Cognifide Limited
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.cognifide.aemrules.htl.visitors;
+
+import com.google.common.collect.Lists;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+import org.apache.sling.scripting.sightly.compiler.expression.Expression;
+import org.apache.sling.scripting.sightly.impl.compiler.Syntax;
+import org.apache.sling.scripting.sightly.impl.compiler.frontend.ExpressionParser;
+import org.apache.sling.scripting.sightly.impl.compiler.frontend.Fragment;
+import org.apache.sling.scripting.sightly.impl.compiler.frontend.Interpolation;
+import org.sonar.plugins.html.node.Attribute;
+import org.sonar.plugins.html.node.CommentNode;
+import org.sonar.plugins.html.node.DirectiveNode;
+import org.sonar.plugins.html.node.ExpressionNode;
+import org.sonar.plugins.html.node.Node;
+import org.sonar.plugins.html.node.TagNode;
+import org.sonar.plugins.html.node.TextNode;
+import org.sonar.plugins.html.visitor.HtmlSourceCode;
+
+public class HtlScanner {
+
+    private static final ExpressionParser expressionParser = new ExpressionParser();
+
+    private final List<DefaultHtlVisitor> metricVisitors;
```

Comment:
What is the purpose of metricVisitors and where are there used outside this class?

Reply:
removed